erpnext/stock/doctype/delivery_note/delivery_note.py

```diff
@@ -657,6 +661,24 @@ def make_sales_invoice(source_name, target_doc=None):
 	invoiced_qty_map = get_invoiced_qty_map(source_name)
 
 	def set_missing_values(source, target):
+		try:
+			parent_payment_term_templates = frappe.get_all(
+				"Sales Order",
+				{
+					"name": ["IN", [d.against_sales_order for d in doc.items]],
```

Comment:
I've been trying to test use cases where d.against_sales_order is NULL in all rows. Expecting pymysql to throw an exception. But seems this is being handled somewhere in frappe maybe?

Reply:
The below query will be generated if there is no `against_sales_order`.

```
select
    `payment_terms_template`
from
    `tabSales Order`
where
    coalesce(`tabSales Order`.`name`, '') IN ('')
order by
    `tabSales Order`.docstatus asc,
    `tabSales Order`.`modified` DESC
```

A check before would be good:

```
against_sales_orders = [d.against_sales_order for d in doc.items if d.against_sales_order]

if against_sales_orders:
    parent_payment_term_templates = frappe.get_all(...
```